### ISSUE TITLE
Store operation not storing image container in disk entry

### ISSUE THREAD

**Issue by ocastx** [twitter/ios-twitter-image-pipeline#11]
While debugging an issue with an particular image not storing in the pipeline (generally no entry in all cache levels) I stumbled over a possible bug in the code.

In the TIPImageStoreOperation main method the cache entries for the different caches are created by deciding which source of data should be held by the entry. 
In the method the memoryEntry is created and it holds the imageContainer, if it exists, in the completeImage property. After that, when the diskEntry is created and the imageFilePath as well as the imageData does not exist, the imageContainer should be stored in the diskEntry. Currently the imageContainer will then be stored in the memoryEntry.

```objective-c
// Create Disk Entry
    TIPImageCacheEntry *diskEntry = nil;
    if (_pipeline.diskCache) {
        diskEntry = [[TIPImageCacheEntry alloc] init];
        if (imageFilePath && ([[NSFileManager defaultManager] fileExistsAtPath:imageFilePath] || (!imageData && !imageContainer))) {
            diskEntry.completeImageFilePath = imageFilePath;
        } else if (imageData) {
            diskEntry.completeImageData = imageData;
        } else {
            TIPAssert(imageContainer);
            memoryEntry.completeImage = imageContainer;
            TIPAssert(memoryEntry.completeImage);
        }
    }
```

I don't know if I'm overlooking something which is definitely possible but it looks like a copy and paste error?

Either way I think it doesn't solve my issue but it's worth looking into 😄

**Comment by ocastx** [comment]
Actually I just found out that this bug is related to the bug that I thought was in my code.

I tried saving two images captured, the first one is the original raw image image from a camera and the second one is a smaller edited version. The edited version was always saved in the pipeline and the original one wasn't. 
After debugging through the framework I found out, that the image was too big for the memory because of the boundaries set by the configuration. Although the image is too big it should be saved to the disk, which wasn't done because of the above mentioned bug. After using the updated code, both images were saved to the pipeline as expected.

One thought, which could have helped finding this bug and could maybe also help finding bottlenecks, is a debug message in ``` updateImageEntry:forciblyReplaceExisting:``` in TPImageMemoryCache that the memory cache was skipped due to max size of the cache entry.

Now after finding this one I must say I'm very happy using this library as it is well written and it really helps in my current/future projects. :)
